support/timepb/cmp.go

```diff
@@ -0,0 +1,90 @@
+package timepb
+
+import (
+	"fmt"
+	"time"
+
+	durpb "google.golang.org/protobuf/types/known/durationpb"
+	tspb "google.golang.org/protobuf/types/known/timestamppb"
+)
+
+// IsZero returns true when t is nil or is zero unix timestamp (1970-01-01)
+func IsZero(t *tspb.Timestamp) bool {
+	return t == nil || t.Nanos == 0 && t.Seconds == 0
+}
+
+// Commpare t1 and t2 and returns -1 when t1 < t2, 0 when t1 == t2 and 1 otherwise.
+// Returns false if t1 or t2 is nil
+func Compare(t1, t2 *tspb.Timestamp) int {
+	if t1 == nil || t2 == nil {
+		panic(fmt.Sprint("Can't compare nil time, t1=", t1, "t2=", t2))
+	}
+	if t1.Seconds == t2.Seconds && t1.Nanos == t2.Nanos {
+		return 0
+	}
+	if t1.Seconds < t2.Seconds || t1.Seconds == t2.Seconds && t1.Nanos < t2.Nanos {
+		return -1
+	}
+	return 1
+}
+
+func DurationIsNegative(d durpb.Duration) bool {
+	return d.Seconds < 0 || d.Seconds == 0 && d.Nanos < 0
```

Comment:
not sure how durations are calculated between seconds + nanos, but would the following invalidate this?:

```go
	d := dur.Duration{
		Seconds: 1,
		Nanos:   -100000000000000000000000,
	}
```

also my IDE is complaining that the parameter should be a pointer

Reply:
This is invalid duration, it won't pass `CheckValid()`.
I don't think we should care with invalid values here, they should be checked upfront.

Reply:
> also my IDE is complaining that the parameter should be a pointer

I'm curious why? `durpb.New` indeed returns a pointer, but he structure is small so there is no need to use pointers. But I will update and use pointer in this function.

Reply:
On the other hand it feels weird to use `time.Duration` without pointer and `durpb.Durration` with pointer.

Reply:
nice.
I've already updated to use pointers.